### PR TITLE
feat!: Use nock@14, support node@20/fetch

### LINKED ISSUE #77: Node 18 and @octokit/core v5.0.0 incompatability
**Describe the bug**
`Moctokit` does not work with Node 18. All `.reply` responses return the same result to Octokit. Setting the `status` as `200` and `404` do not differ the mock response`

I believe this is due to [Node 18 using a new http client](https://nodejs.org/en/blog/announcements/v18-release-announce) named [undici](https://github.com/nodejs/undici) for `fetch` API support

Looking at the code for Moctokit it uses `nock`, which has [an open issue for undici support](https://github.com/nock/nock/issues/2183)

**To Reproduce**
Use `Moctokit` with Node 18 and `@octokit/core` `v5.0.0`

**Expected behavior**
Proper mock responses are returned.

**Additional context**
I don't actually expect this to be solved until Nock is updated but an incompatibility note in the `README` would be nice.

### PR DESCRIPTION
**Pull request by oliversalzburg**
The `toStrictEqual` calls had to be replaced, because it falsely assumed to receive back the exact same in-memory object, which seems like an unrealistic expectation in the test, that was only satisfied through the previous dependency implementation details.

Fixes #77

### CODE REVIEW

**Comment by shubhbapna** [comment]
Thank you for your PR! I am currently on vacation and will be back on 12th June

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
         with:
-          node-version: 16
+          node-version: 22
```
Comment (shubhbapna) [inline .github/workflows/publish.yml:17]: The current LTS is node 20 so I would change it to that (and in all the other workflows as well)

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
   tests:
     strategy:
       matrix:
-        node-version: [16.x]
+        node-version: [20,22]
```
Comment (shubhbapna) [inline .github/workflows/pull_request.yml:16]: Could you add node 18 to the list as well

File: package.json [hunk 3]
```diff
@@ -68,10 +68,13 @@
     "express": "^4.18.1",
     "fast-glob": "^3.2.12",
     "fs-extra": "^10.1.0",
-    "nock": "^13.2.7",
+    "nock": "14.0.0-beta.7",
     "simple-git": "^3.8.0",
     "totalist": "^3.0.0"
   },
+  "engines": {
+    "node": "20"
```
Comment (shubhbapna) [inline package.json:76]: Could you change this to node 18

**Review by shubhbapna — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (oliversalzburg) [reply package.json]: I realized that this is too strict anyway, and fixed this as `>=18`.

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -14,7 +14,7 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
         with:
-          node-version: 16
+          node-version: 20
```
Comment (shubhbapna) [inline .github/workflows/publish.yml:17]: Should be node 18 since that is the minimum supported version

File: .github/workflows/pull_request.yml [hunk 5]
```diff
@@ -36,7 +36,7 @@ jobs:
 
       - uses: actions/setup-node@v3
         with:
-          node-version: 16.x
+          node-version: 22
```
Comment (shubhbapna) [inline .github/workflows/pull_request.yml:39]: Should be node 18

File: .github/workflows/report.yml [hunk 6]
```diff
@@ -18,7 +18,7 @@ jobs:
         uses: warchant/setup-sonar-scanner@v3     
       - uses: actions/setup-node@v3
         with:
-          node-version: 16
+          node-version: 22
```
Comment (shubhbapna) [inline .github/workflows/report.yml:21]: Should be node 18

File: .github/workflows/update.yml [hunk 7]
```diff
@@ -14,7 +14,7 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
         with:
-          node-version: 16
+          node-version: 22
```
Comment (shubhbapna) [inline .github/workflows/update.yml:17]: Should be node 18

**Review by shubhbapna — CHANGES_REQUESTED** [review]

**Review by shubhbapna — APPROVED** [review]

**Comment by oliversalzburg** [comment]
> So sorry for the delay, working on this and should have a beta release available by EOD

I really appreciate that you took the time to evaluate this. I only hope that we can make this amazing project work with the latest libraries.

**Pull request merged**